### PR TITLE
BUG - MULTICAMPUS: User can access a session that is assigned to another campus in which the user is not registered

### PR DESCRIPTION
**Pull request by juan-cortizas-ponte**
Enviroment: Multicampus configuration
Steps:

1. Go to a chamilo multicampus site, for example site.com
2. Go to the campus catalog: site.com/main/auth/courses.php
3. Select one of the sessions and go to to its "about" page, for example: site.com/session/{session_id}/about/
4. Change the session id of the url for one of an existing session assigned to another campus
5. User can see the info of the session of the other campus an enroll in it

This PR adds a new function to SessionManager to allow check if a session is assigned to a access url

This new function is used on main/session/about.php to check if the session is assigned to the current access url

### CODE REVIEW

File: main/inc/lib/sessionmanager.lib.php [hunk 1]
```diff
@@ -9919,6 +9919,32 @@ public static function importAgendaFromSessionModel(int $modelSessionId, int $se
...
+                COUNT(*) as count
+            FROM $session_table s
+            INNER JOIN $table_access_url_rel_session ar ON ar.session_id = s.id
+            WHERE s.id = $sessionId AND ar.access_url_id = $urlId ";
+
+        $result = Database::fetch_array(Database::query($sql));
+
+        return $result['count'] > 0;
+    }
+
```
Comment (AngelFQC) [inline main/inc/lib/sessionmanager.lib.php:9947]: Ya existe el método UrlManager::relation_url_session_exist

https://github.com/chamilo/chamilo-lms/blob/b60be692811d61f383c3b3857707a36b6b3be247/main/inc/lib/urlmanager.lib.php#L502-L515

Comment (juan-cortizas-ponte) [reply main/inc/lib/sessionmanager.lib.php:9947]: Muchas gracias @AngelFQC !, desconocía ese método, nuevo push para eliminar el método extra creado y hacer uso de UrlManager::relation_url_session_exist

**Comment by ywarnier** [comment]
@juan-cortizas-ponte no me queda claro porqué eliminas `compareByCourse` pero no `sessionIsAssignedToUrl`... ? Supongo que se trata de un error de manipulación.

**Comment by juan-cortizas-ponte** [comment]
> @juan-cortizas-ponte no me queda claro porqué eliminas `compareByCourse` pero no `sessionIsAssignedToUrl`... ? Supongo que se trata de un error de manipulación.

Totalmente un error grave por mi parte al ir en modo "piloto automático", eliminé el método compareByCourse en vez de sessionIsAssignedToUrl.

He revertido el commit 55974b0d54f6e641649e31eb930f4a89c180b6a4 y creado uno nuevo eliminando sessionIsAssignedToUrl y usando UrlManager::relation_url_session_exist

**Comment by ywarnier** [comment]
Pues esto lo hace fácil de integrar ahora :smile:

**Pull request merged**